src/mainwindow.cpp

```diff
@@ -80,6 +80,20 @@ MainWindow::MainWindow(QWidget* parent):
     viewModeGroup->addAction(ui_->actionDirTreeMode);
     viewModeGroup->addAction(ui_->actionFlatListMode);
 
+    // add fallback icons for icon sets using only current Freedesktop standard icon names
+    // NOTE: ui_->actionXyz->icon().isNull() returns false though the icon is not visible,
+    // so can't check from UI (Qt 5.15.10)
+    ui_->actionExtract->setIcon(QIcon::fromTheme(QStringLiteral("archive-extract"),
+                                                 QIcon::fromTheme(QStringLiteral("go-up"))));
+
+    ui_->actionAddFiles->setIcon(
+        QIcon::fromTheme(QStringLiteral("archive-insert"),
+                         QIcon::fromTheme(QStringLiteral("insert-object"))));
+
+    ui_->actionAddFolder->setIcon(
+        QIcon::fromTheme(QStringLiteral("archive-insert-directory"),
+                         QIcon::fromTheme(QStringLiteral("insert-object"))));
+
```

Comment:
Thanks!

Please remove the icons from the UI file, and change the long comment to something like

```c++
    // icons are set here for having fallbacks
```

Also, all instances of `setIcon` can be in two lines (like the first one, which is the cleanest), and the empty lines between them can be removed for grouping them together.

Reply:
Should be OK, I'm used to QtC clang-format plugin.